### PR TITLE
Add in focal to unstable and non-upgrade tests

### PR DESCRIPTION
**Pull request by amanda11**

### CODE REVIEW

**Comment by amanda11** [comment]
Updating st2ci to include focal. Have not included upgrade or stable runs as can't do that until we start the release prep.
I thought that the PR runs were disabled, but not sure if that has been done directly on the CI server rather than on the pack - so would be good to clarify if the st2 e2e on prs should be disabled not enabled in pack for now.

File: actions/workflows/run_all_nightlies.yaml [hunk 1]
```diff
@@ -9,16 +9,24 @@ input:
   - debug
 
 vars:
+# Can put back to just distros once U20 is released
```
Comment (blag) [inline actions/workflows/run_all_nightlies.yaml:12]:
```suggestion
# TODO: Uncomment UBUNTU20 once we have released ST2 3.4 with Ubuntu 20.04 support
```

File: actions/workflows/run_all_nightlies.yaml [hunk 2]
```diff
@@ -9,16 +9,24 @@ input:
...
       RHEL8: el8
       UBUNTU16: u16
       UBUNTU18: u18
+      UBUNTU20: u20
+  - stabledistros:
+      RHEL7: el7
+      RHEL8: el8
+      UBUNTU16: u16
+      UBUNTU18: u18
+
```
Comment (blag) [inline actions/workflows/run_all_nightlies.yaml:24]:
```suggestion
      # UBUNTU20: u20
```

**Review by blag — CHANGES_REQUESTED** [review]
Instead of creating a new key, I would just add a new item and comment it out.

File: actions/workflows/run_all_nightlies.yaml (hunk 2 above)
Comment (blag) [reply actions/workflows/run_all_nightlies.yaml:24]: You'll also need to revert where `ctx().stabledistros` is used back to just `ctx().distros`.

Comment (amanda11) [reply actions/workflows/run_all_nightlies.yaml:24]: So your suggestion of just keeping them separate and having it commented out is easier.

File: actions/workflows/run_all_nightlies.yaml [hunk 3]
```diff
@@ -9,16 +9,18 @@ input:
...
       RHEL8: el8
       UBUNTU16: u16
       UBUNTU18: u18
+      # UBUNTU20: u20
 
 tasks:
   init:
     with:
-      items: distro, abbr in <% zip(ctx().distros.keys(), ctx().distros.values()) %>
+      items: distro, abbr in <% zip(ctx().stabledistros.keys(), ctx().stabledistros.values()) %>
```
Comment (blag) [inline actions/workflows/run_all_nightlies.yaml:23]: Don't forget to set these back.

File: actions/workflows/run_all_nightlies.yaml [hunk 4]
```diff
@@ -29,8 +31,8 @@ tasks:
 
   st2_pkg_test_stable_all:
     with:
-      items: distro, abbr in <% zip(ctx().distros.keys(), ctx().distros.values()) %>
-      concurrency: <% len(ctx().distros) %>
+      items: distro, abbr in <% zip(ctx().stabledistros.keys(), ctx().stabledistros.values()) %>
+      concurrency: <% len(ctx().stabledistros) %>
```
Comment (blag) [inline actions/workflows/run_all_nightlies.yaml:35]: Don't forget to set these back.

**Comment by amanda11** [comment]
@blag Are you ok to re-review? I've gone for keeping the two separate lists one for stable and one for unstable, then we just need to uncomment the U20 on the stable list when we release 3.5. Makes it easier in future to add or remove from the stable/unstable checks when we add/remove OSes.

File: actions/workflows/run_all_nightlies.yaml (hunk 3 above)
Comment (blag) [inline actions/workflows/run_all_nightlies.yaml:23]:
```suggestion
      items: distro, abbr in <% zip(ctx().distros.keys(), ctx().distros.values()) %>
```

File: actions/workflows/run_all_nightlies.yaml (hunk 4 above)
Comment (blag) [inline actions/workflows/run_all_nightlies.yaml:35]:
```suggestion
      items: distro, abbr in <% zip(ctx().distros.keys(), ctx().distros.values()) %>
      concurrency: <% len(ctx().distros) %>
```

**Review by blag — CHANGES_REQUESTED** [review]
Still need to revert some changes - see my suggestions.

**Comment by blag** [comment]
After my suggested changes are merged, I'm feeling pretty confident about this PR. I would be fine with it if you wanted to merge this and just redeploy `master` to st2cicd. If we need to fix anything we an do that in a separate PR. That's a choice I leave to you.

**Comment by amanda11** [comment]
@blag As discussed on slack, I think we need the two lists, if you are ok with that can you approve.
The PR is on st2cicd at the moment, and correctly running focal just for unstable and not stable builds.

**Review by blag — APPROVED** [review]

**Pull request merged**